resources: Update GPU resources to Vega

abmerop opened the pull request:
This PR is part of [gem5/#703](https://github.com/gem5/gem5/issues/703) to update documentation. These commits do the following:

- Replace GCN3_X86 with VEGA_X86 for build and run commands.
- Replace gfx801 and gfx803 with gfx902 and gfx900 respectively.
- Update pre-built binary website paths to a v24-0 folder. These will have to be rebuilt to removed gfx8 targets and add gfx902.
- Update the docker version in READMEs to v23-1. This docker contains the additional packages and env variables to build HACC.
- Remove the hsa-agent-pkt application. This application has been broken for years and only builds with ROCm 1.6, so assuming it is no longer used.

abmerop committed:
resources: Update GCN3/gfx8 to VEGA/gfx9

The commit does a few things related to simple renames of GCN3. GCN3 is
replaced with Vega in all locations. The targets gfx801 and gfx803 are
replaced with the corresponding gfx9 APU and dGPU targets (gfx902 and
gfx900 respectively). The docker commands update v22-1 with v23-1. This
newer docker image contains an MPI install so that HACC can be built
without a separate docker image. The location of the prebuilt binaries
is updated to a v24-0 path. The old paths contain binaries built with
gfx801, gfx802, and gfx900. They are therefore missing gfx902 and should
be rebuilt before the next release.

abmerop committed:
resources: Remove hsa-agent-pkt resource

This application has not been updated with bump to ROCm 4.0 in GPU SE
mode. The code is still written in ROCm 1.6 style and therefore does not
compile and there are include files missing. Not sure what the purpose
of this resource is, so removing it in future releases.

abmerop commented:
Will add another commit to this to fix the pannotia instructions but the commits I have now are ready for comments.  Currently they do not work due to docker not allowing referencing the graph parser up one directory and not being able to find m5ops.h.

mattsinc commented:
> Will add another commit to this to fix the pannotia instructions but the commits I have now are ready for comments. Currently they do not work due to docker not allowing referencing the graph parser up one directory and not being able to find m5ops.h.

Yeah ... that is a known issue and I thought Kai had updated the instructions for that, but I guess not ...

mattsinc commented on README.md:
````diff
@@ -341,46 +341,20 @@ git clone https://gem5.googlesource.com/public/gem5-resources
-```
-
-The compiled binary can be found in `src/gpu/hsa-agent-pkt/bin`
+<http://dist.gem5.org/dist/v24-0/test-progs/square/square>
````
Have you confirmed this path works/exists?

powerjg replied:
@Harshil2107 can you build this binary and upload it to this location? If you can't figure out how to build it from this documentation, please reach out to Matt P.

abmerop replied:
Yea, these don't exist yet.  I put this on develop as GCN3 isn't yet removed but should be by v24-0 and therefore it makes sense to put the binaries in a v24-0 folder.

I updated the docker image to build the applications which will need to be pulled first in order to build the gfx902 target: https://github.com/gem5/gem5/pull/804 .  Maybe we should wait for that to be built first? Or Harshil can build it locally using the Dockerfile from that PR.

mattsinc replied:
Yeah, I'll defer to @powerjg and @Harshil2107 about what order to push things, but rest of this makes sense.

mattsinc commented on README.md:
```diff
@@ -397,26 +371,26 @@ docker run --rm -v ${PWD}:${PWD} -w ${PWD} -u $UID:$GID ghcr.io/gem5/gcn-gpu mak
 Individual programs can be made by specifying the name of the program
 
-By default, this code builds for gfx801, a GCN3-based APU. This can be
+By default, this code builds for gfx901, a VEGA-based APU. This can be
```
gfx902?

abmerop committed:
resources: Delete HACC docker and update README

The contents of the HACC docker file have been integrated into gcn-gpu
docker v23-1+ and therefore the additional docker image is no longer
necessary. The Dockerfile is deleted and the README is updated to be
similar to the READMEs for other GPU applications.

powerjg commented on README.md:
```diff
@@ -397,26 +371,26 @@ docker run --rm -v ${PWD}:${PWD} -w ${PWD} -u $UID:$GID ghcr.io/gem5/gcn-gpu mak
 ## Pre-built binary
 
-<http://dist.gem5.org/dist/v22-0/test-progs/hip-samples/2dshfl>
+<http://dist.gem5.org/dist/v24-0/test-progs/hip-samples/2dshfl>
```
Same for these @Harshil2107

abmerop replied:
Similarly, I updated the docker image to build the applications which will need to be pulled first in order to build the gfx902 target: https://github.com/gem5/gem5/pull/804 or wait until it is on ghcr.

powerjg commented on src/gpu/pannotia/color/README.md:
````diff
@@ -31,34 +31,34 @@ cd src/gpu/pannotia/clr
+Color is a GPU application, which requires that gem5 is built with the VEGA_X86 (or Vega_X86, although this has been less heavily tested) architecture.  The test is run with the VEGA_X86 gem5 variant, compiled using the gcn-gpu docker image:
 
 ```
 git clone https://gem5.googlesource.com/public/gem5
````
github :). Should probably do a search for anything else that's googlesource.

abmerop replied:
Good catch... Changed all of the GPU related ones.  There's many googlesource references in the main README.md but I haven't modified those

powerjg commented on src/gpu/pannotia/mis/README.md:
```diff
@@ -6,12 +6,12 @@ tags:
 Maximal Independent Set (mis) is a graph analytics application that is part of the Pannotia benchmark suite.  It is designed to find a maximal subset of vertices in a graph such that no two are adjacent.  The provided version is for use with the gpu-compute model of gem5.  Thus, it has been ported from the prior CUDA and OpenCL variants to HIP, and validated on a Vega-class AMD GPU.
 
-Compiling MIS, compiling the GCN3_X86/Vega_X86 versions of gem5, and running MIS on gem5 is dependent on the gcn-gpu docker image, `util/dockerfiles/gcn-gpu/Dockerfile` on the [gem5 stable branch](https://gem5.googlesource.com/public/gem5/+/refs/heads/stable).
+Compiling MIS, compiling the VEGA_X86/Vega_X86 versions of gem5, and running MIS on gem5 is dependent on the gcn-gpu docker image, `util/dockerfiles/gcn-gpu/Dockerfile` on the [gem5 stable branch](https://gem5.googlesource.com/public/gem5/+/refs/heads/stable).
```
github, not googlesource

powerjg reviewed (commented):
I'm good with pushing this as-is. We can make further improvements later. This assumes that you've addressed Matt S.'s comments.

Harshil2107 commented on src/gpu/square/README.md:
````diff
@@ -1,44 +1,44 @@
 
 ```
-docker run -u $UID:$GID --volume $(pwd):$(pwd) -w $(pwd) ghcr.io/gem5/gcn-gpu:v22-1 gem5/build/GCN3_X86/gem5.opt gem5/configs/example/apu_se.py -n 3 -c bin/square
+docker run -u $UID:$GID --volume $(pwd):$(pwd) -w $(pwd) ghcr.io/gem5/gcn-gpu:v23-1 gem5/build/VEGA_X86/gem5.opt gem5/configs/example/apu_se.py -n 3 -c bin/square
````
I get the following error after trying to run the square script. I built a new binary following the steps of  the updated readme.

I get the following warning when I try to build the image, I am using the following docker file (https://github.com/gem5/gem5/blob/bae34876780dfb2bc22b9151bfda1d39ee80cfb1/util/dockerfiles/gcn-gpu/Dockerfile):
```
/opt/rocm/hip/bin/hipcc -DGFX9 --amdgpu-target=gfx900,gfx902 src/main.hip.cpp -o bin/allSyncPrims-1kernel
Warning: The specified HIP target: gfx902 is unknown. Correct compilation is not guaranteed.
```
The following is the error when I run the simulation:
```
sudo docker run -u $UID:$GID --volume $(pwd):$(pwd) -w $(pwd) ghcr.io/gem5/gcn-gpu:v23-1 gem5/build/VEGA_X86/gem5.opt gem5/configs/example/apu_se.py -n 3 -c bin/square
Global frequency set at 1000000000000 ticks per second
src/mem/dram_interface.cc:690: warn: DRAM device capacity (8192 Mbytes) does not match the address range assigned (512 Mbytes)
src/base/statistics.hh:279: warn: One of the stats is a legacy stat. Legacy stat is a stat that does not belong to any statistics::Group. Legacy stat is deprecated.
src/base/stats/storage.hh:278: warn: Bucket size (5) does not divide range [1:75] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (2) does not divide range [1:10] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (2) does not divide range [1:64] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (5) does not divide range [1:75] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (2) does not divide range [1:10] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (2) does not divide range [1:64] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (5) does not divide range [1:75] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (2) does not divide range [1:10] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (2) does not divide range [1:64] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (5) does not divide range [1:75] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (2) does not divide range [1:10] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (2) does not divide range [1:64] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/statistics.hh:279: warn: One of the stats is a legacy stat. Legacy stat is a stat that does not belong to any statistics::Group. Legacy stat is deprecated.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
src/base/stats/storage.hh:278: warn: Bucket size (10000) does not divide range [1:1.6e+06] into equal-sized buckets. Rounding up.
Forcing maxCoalescedReqs to 32 (TLB assoc.) 
Forcing maxCoalescedReqs to 32 (TLB assoc.) 
Forcing maxCoalescedReqs to 32 (TLB assoc.) 
Forcing maxCoalescedReqs to 32 (TLB assoc.) 
Forcing maxCoalescedReqs to 32 (TLB assoc.) 
Forcing maxCoalescedReqs to 32 (TLB assoc.) 
src/base/statistics.hh:279: warn: One of the stats is a legacy stat. Legacy stat is a stat that does not belong to any statistics::Group. Legacy stat is deprecated.
Forcing maxCoalescedReqs to 32 (TLB assoc.) 
Forcing maxCoalescedReqs to 32 (TLB assoc.) 
src/base/remote_gdb.cc:418: warn: Sockets disabled, not accepting gdb connections
warn: system.ruby.network adopting orphan SimObject param 'ext_links'
warn: system.ruby.network adopting orphan SimObject param 'int_links'
src/sim/simulate.cc:199: info: Entering event queue @ 0.  Starting simulation...
gem5 Simulator System.  https://www.gem5.org
gem5 is copyrighted software; use the --copyright option for details.

gem5 version 23.1.0.0
gem5 compiled Jan 30 2024 23:01:35
gem5 started Jan 30 2024 23:04:53
gem5 executing on 00999f687fc7, pid 1
command line: gem5/build/VEGA_X86/gem5.opt gem5/configs/example/apu_se.py -n 3 -c bin/square

info: Standard input is not a terminal, disabling listeners.
Num SQC =  1 Num scalar caches =  1 Num CU =  4
src/mem/ruby/system/Sequencer.cc:668: warn: Replacement policy updates recently became the responsibility of SLICC state machines. Make sure to setMRU() near callbacks in .sm files!
src/sim/mem_state.cc:448: info: Increasing stack size by one page.
src/base/loader/symtab.cc:95: warn: Cannot insert a new symbol table due to name collisions. Adding prefix to each symbol's name can resolve this issue.
src/base/loader/symtab.cc:95: warn: Cannot insert a new symbol table due to name collisions. Adding prefix to each symbol's name can resolve this issue.
src/base/loader/symtab.cc:95: warn: Cannot insert a new symbol table due to name collisions. Adding prefix to each symbol's name can resolve this issue.
src/base/loader/symtab.cc:95: warn: Cannot insert a new symbol table due to name collisions. Adding prefix to each symbol's name can resolve this issue.
src/sim/syscall_emul.cc:74: warn: ignoring syscall mprotect(...)
src/sim/syscall_emul.cc:74: warn: ignoring syscall mprotect(...)
src/base/loader/symtab.cc:95: warn: Cannot insert a new symbol table due to name collisions. Adding prefix to each symbol's name can resolve this issue.
src/sim/syscall_emul.cc:74: warn: ignoring syscall mprotect(...)
src/base/loader/symtab.cc:95: warn: Cannot insert a new symbol table due to name collisions. Adding prefix to each symbol's name can resolve this issue.
src/base/loader/symtab.cc:95: warn: Cannot insert a new symbol table due to name collisions. Adding prefix to each symbol's name can resolve this issue.
src/base/loader/symtab.cc:95: warn: Cannot insert a new symbol table due to name collisions. Adding prefix to each symbol's name can resolve this issue.
src/base/loader/symtab.cc:95: warn: Cannot insert a new symbol table due to name collisions. Adding prefix to each symbol's name can resolve this issue.
src/sim/syscall_emul.cc:74: warn: ignoring syscall mprotect(...)
src/base/loader/symtab.cc:95: warn: Cannot insert a new symbol table due to name collisions. Adding prefix to each symbol's name can resolve this issue.
src/base/loader/symtab.cc:95: warn: Cannot insert a new symbol table due to name collisions. Adding prefix to each symbol's name can resolve this issue.
src/sim/syscall_emul.cc:74: warn: ignoring syscall mprotect(...)
src/base/loader/symtab.cc:95: warn: Cannot insert a new symbol table due to name collisions. Adding prefix to each symbol's name can resolve this issue.
src/sim/syscall_emul.cc:74: warn: ignoring syscall mprotect(...)
src/sim/syscall_emul.cc:74: warn: ignoring syscall mprotect(...)
src/sim/syscall_emul.cc:74: warn: ignoring syscall mprotect(...)
src/sim/syscall_emul.cc:74: warn: ignoring syscall mprotect(...)
src/sim/syscall_emul.cc:74: warn: ignoring syscall mprotect(...)
src/sim/syscall_emul.cc:74: warn: ignoring syscall mprotect(...)
src/sim/syscall_emul.cc:74: warn: ignoring syscall mprotect(...)
src/sim/syscall_emul.cc:74: warn: ignoring syscall mprotect(...)
src/sim/syscall_emul.cc:74: warn: ignoring syscall mprotect(...)
src/sim/syscall_emul.cc:74: warn: ignoring syscall mprotect(...)
src/sim/syscall_emul.cc:74: warn: ignoring syscall mprotect(...)
src/sim/syscall_emul.cc:74: warn: ignoring syscall mprotect(...)
src/sim/syscall_emul.cc:74: warn: ignoring syscall mprotect(...)
src/sim/syscall_emul.cc:74: warn: ignoring syscall mprotect(...)
src/sim/syscall_emul.cc:74: warn: ignoring syscall mprotect(...)
src/sim/syscall_emul.cc:74: warn: ignoring syscall mprotect(...)
src/sim/syscall_emul.cc:74: warn: ignoring syscall mprotect(...)
src/sim/syscall_emul.cc:74: warn: ignoring syscall mprotect(...)
src/sim/syscall_emul.cc:74: warn: ignoring syscall mprotect(...)
src/sim/syscall_emul.cc:74: warn: ignoring syscall set_robust_list(...)
src/sim/syscall_emul.cc:85: warn: ignoring syscall rt_sigaction(...)
      (further warnings will be suppressed)
src/sim/syscall_emul.cc:85: warn: ignoring syscall rt_sigprocmask(...)
      (further warnings will be suppressed)
src/sim/syscall_emul.cc:74: warn: ignoring syscall get_mempolicy(...)
build/VEGA_X86/arch/x86/generated/exec-ns.cc.inc:27: warn: instruction 'frndint' unimplemented
src/sim/mem_state.cc:448: info: Increasing stack size by one page.
src/gpu-compute/gpu_compute_driver.cc:710: warn: unimplemented ioctl: AMDKFD_IOC_ACQUIRE_VM
src/sim/syscall_emul.hh:1890: warn: mmap: writing to shared mmap region is currently unsupported. The write succeeds on the target, but it will not be propagated to the host or shared mappings
src/sim/mem_state.cc:448: info: Increasing stack size by one page.
src/gpu-compute/gpu_compute_driver.cc:460: warn: Signal events are only supported currently
src/sim/syscall_emul.cc:74: warn: ignoring syscall mprotect(...)
src/sim/power_state.cc:105: warn: PowerState: Already in the requested power state, request ignored
src/sim/syscall_emul.cc:74: warn: ignoring syscall set_robust_list(...)
src/sim/syscall_emul.cc:74: warn: ignoring syscall mprotect(...)
src/gpu-compute/gpu_compute_driver.cc:604: warn: unimplemented ioctl: AMDKFD_IOC_SET_SCRATCH_BACKING_VA
src/gpu-compute/gpu_compute_driver.cc:614: warn: unimplemented ioctl: AMDKFD_IOC_SET_TRAP_HANDLER
/HIP/rocclr/hip_code_object.cpp:120: guarantee(false && "hipErrorNoBinaryForGpu: Coudn't find binary for current devices!")
Ticks: 72699332000
Exiting because  exiting with last active thread context
```

I followed the following steps:
```
> cd src/gpu/square
> sudo docker run --rm -v ${PWD}:${PWD} -w ${PWD} -u $UID:$GID ghcr.io/gem5/gcn-gpu:v23-1 make
> git clone https://github.com/gem5/gem5
> cd gem5
> sudo docker run -u $UID:$GID --volume $(pwd):$(pwd) -w $(pwd) ghcr.io/gem5/gcn-gpu:v23-1 scons build/VEGA_X86/gem5.opt -j 20
> sudo docker run -u $UID:$GID --volume $(pwd):$(pwd) -w $(pwd) ghcr.io/gem5/gcn-gpu:v23-1 gem5/build/VEGA_X86/gem5.opt gem5/configs/example/apu_se.py -n 3 -c bin/square
```

Harshil2107 replied:
I also get the same error when I run the newly built heterosync binary.

powerjg replied:
It looks like ghcr.io/gem5/gcn-gpu:v23-1 doesn't have support for gfx902. We're using the docker image built with the dockerfile here: https://github.com/gem5/gem5/blob/stable/util/dockerfiles/gcn-gpu/Dockerfile

Please let us know if this isn't the right docker image to use, and point us to an updated dockerfile if this is the wrong docker image.

powerjg replied:
Sorry if this is causing multiple emails, but this is high priority as it is blocking us fixing the nightly tests: https://github.com/gem5/gem5/actions/runs/7707607338/job/21005088142

mattsinc replied:
I am pretty sure, but not 100% sure, that you'll need to add --gfx-version=gfx902 to the command line (somewhere after apu_se.py) to get this to run as is.  The part I'm not sure about is why @abmerop's changes to make 902 the default did not make it optional for this to be added (maybe that patch hasn't been merged in in the gem5 part of the repo yet?).  But that is the first thing I'd try.

mattsinc replied:
regarding that Dockerfile, @powerjg : it does not have the change in this patch @abmerop pushed -- I believe Matt's commit message says that this fix is needed to (specifically to line 107).  But I'm not sure why Matt's changes to develop would be affecting stable in general -- unless the changes to run gfx902 instead are being sent directly to stable instead of develop somewhere and I missed it?

powerjg replied:
This is not affecting stable, it's the daily tests of develop that's failing.

I know it's a significant amount of work, and I appreciate what you and Matt P. are doing, but the community needs more help in making sure that we can run/use/test the GPU models. Just because it works on your machine doesn't mean that it works for others. The daily tests failing show this.

Let's try to dig into what's going on:
1. When we run the `square` binary that is we previously built (found on dist.gem5.org), we get the error "/HIP/rocclr/hip_code_object.cpp:120: guarantee(false && "hipErrorNoBinaryForGpu: Coudn't find binary for current devices!")" My understanding is that this error means the binary was not compiled for the correct GPU version.
2. When we recompile the square binary, we get the warning "Warning: The specified HIP target: gfx902 is unknown. Correct compilation is not guaranteed." My understanding is that this error means that if the GPU version is gfx902 then the binary won't work with the error above.

What we need to know is (1) how to compile the square binary and (2) how to run the square binary. Both of these should use only docker images stored on ghcr.io (they *cannot* use images that are built locally!) and they must use code that is in gem5 develop without any changes. Similarly we *cannot* check out gem5-resources to run tests. We must *only use precompiled binaries* found on dist.gem5.org (and cataloged on resources.gem5.org). (As a side note, we must have detailed documentation which is reproducible to produce these binaries.) 

Our daily tests have been failing for 6 days now. This is unacceptable. Apologies that you were not notified 5 days ago.

We must fix the daily tests ASAP. Otherwise, we end up in a case where we have errors masking errors and it ends up sucking up a huge amount of our time.

mattsinc replied:
My comment about stable was because you linked to the Dockerfile on stable: https://github.com/gem5/gem5/blob/stable/util/dockerfiles/gcn-gpu/Dockerfile.  I assumed you were trying to say this is what was running, but it does not seem that was the case and maybe was a copy paste error.

In regard to the other part, what I said before is still the solution I would try -- above in @Harshil2107 's steps, this was included:

/opt/rocm/hip/bin/hipcc -DGFX9 --amdgpu-target=gfx900,gfx902 src/main.hip.cpp -o bin/allSyncPrims-1kernel

Perhaps I am projecting, but I am assuming this meant square was compiled in a similar way.  I do not see that above so @Harshil2107 let me know if this is not the case.  Regardless, the error is yes because the simulator is trying to run one generation of GPU code but the binary does not have support for that generation.  I'll need to know how exactly @Harshil2107 compiled it.  However, I do also see that @Harshil2107 ran make.  Without this patch, gfx902 would not be compiled.  @powerjg is telling me you are manually compiling with gfx902 though?  If so, then it should be there.  The way to check would be:

$ROCM_HEAD/bin/extractkernel -9 $file_in

(file_in is the binary file)

This will generated a disassembly for the binary and you should be able to grep for gfx902 in there and see code after it.  However, I'm seeing extractkernel is failing for me on my machine for GPU binaries that actually run on the real GPU.  So I'm questioning if it will work for you.  If this doesn't work, you can add --save-temps to the command line when you compile the file like this:

/opt/rocm/hip/bin/hipcc --amdgpu-target=gfx900,gfx902 --save-temps src/square.cpp -o bin/square

Then let me know what .s files you see as output.

Setting that aside, I am still a little confused about the order patches are being pushed in -- maybe I missed something -- but I would guess that until the patches both for resources and gem5 are pushed in, something would fail.  Let me know if I'm missing something.  @abmerop should chime in too though.

Ultimately I think there are three potential sources of the problem:

1.  Without this patch being merged in, gfx902 is not being compiled for square.  @Harshil2107 needs to confirm if this is happening or not.
2. https://github.com/gem5/gem5/blob/develop/configs/example/apu_se.py#L375 seems to show that the default has been updated to gfx902.  So in theory that means the simulator is expecting gfx902 but not getting it.  Regardless, for now I would try --gfx-version=gfx902 on the command line (somewhere after apu_se.py) and see if that helps.
3. https://github.com/gem5/gem5/pull/804 has not been merged in yet and @abmerop is reporting there that this is causing gfx902 to be clobbered.  So my guess is that this #804 patch might need to be merged in first before this resources patch can be merged in.  It may also explain why the default in https://github.com/gem5/gem5/blob/develop/configs/example/apu_se.py#L375 is being overwritten.  Alternatively, it may highlight that the docker needs to be rebuilt before this all will work.

@Harshil2107 let me know especially about 1-3 above and we can proceed from there.

abmerop replied:
The correct docker is here: https://github.com/gem5/gem5/pull/804 it has not been committed yet.

Matt is correct, the docker image either needs to be commited first or built manual to target gfx902.  Once that is available (As maybe v24-0 tag) I can update this document again

Harshil2107 replied:
The steps I followed to build the binaries are as outlined in the readme. I checkout out this PR and followed the following steps:
```
> cd src/gpu/square
> sudo docker run --rm -v ${PWD}:${PWD} -w ${PWD} -u $UID:$GID ghcr.io/gem5/gcn-gpu:v23-1 make
> git clone https://github.com/gem5/gem5
> cd gem5
> sudo docker run -u $UID:$GID --volume $(pwd):$(pwd) -w $(pwd) ghcr.io/gem5/gcn-gpu:v23-1 scons build/VEGA_X86/gem5.opt -j 20
> sudo docker run -u $UID:$GID --volume $(pwd):$(pwd) -w $(pwd) ghcr.io/gem5/gcn-gpu:v23-1 gem5/build/VEGA_X86/gem5.opt gem5/configs/example/apu_se.py -n 3 -c bin/square
```

> 1. Without this patch being merged in, gfx902 is not being compiled for square. @Harshil2107 needs to confirm if this is happening or not.

Yes, I assumed the docker image was updated, I didnot compile gfx902 seperatly

> 3. https://github.com/gem5/gem5/pull/804 has not been merged in yet and @abmerop is reporting there that this is causing gfx902 to be clobbered. So my guess is that this #804 patch might need to be merged in first before this resources patch can be merged in. It may also explain why the default in https://github.com/gem5/gem5/blob/develop/configs/example/apu_se.py#L375 is being overwritten. Alternatively, it may highlight that the docker needs to be rebuilt before this all will work.

I think we should get the docker image in first, this might be a primary issue. I resolved a merge conflict on this PR, if we can get this in, I can update the docker image on the cloud. I am currently building this new docker image locally to see if I can build the binaries locally.

powerjg commented:
We'll try building a new docker image and then using that image to compile the binaries.

It seems there was some serious miscommunication and unknown dependences between changes here. We decided to transition the tests from GCN3 to VEGA, but there were changes that needed to be merged, a docker image that needed to be built and pushed to gchr.io, changes to gem5 resources that needed to be merged, and binaries that needed to be built and pushed to dist.gem5.org before we could transition the tests. Next time, it would be helpful for someone with a full view of the GPU changes needed to communicate all of this.

abmerop commented:
> We'll try building a new docker image and then using that image to compile the binaries.
> 
> It seems there was some serious miscommunication and unknown dependences between changes here. We decided to transition the tests from GCN3 to VEGA, but there were changes that needed to be merged, a docker image that needed to be built and pushed to gchr.io, changes to gem5 resources that needed to be merged, and binaries that needed to be built and pushed to dist.gem5.org before we could transition the tests. Next time, it would be helpful for someone with a full view of the GPU changes needed to communicate all of this.

I did mention the new docker image twice a week ago in this PR

abmerop commented:
I don't know what's being tested right now but it looks like only square and heterosync?  Here are binaries with gfx902 code objects if that's useful while getting the docker up.

http://abmerop.xyz/gem5/v24-0/gpu-bin/square
http://abmerop.xyz/gem5/v24-0/gpu-bin/allSyncPrims-1kernel

Harshil2107 commented:
I made the new image locally and built the square binary. I was able to successfully run the binary with apu_se.py with the new image.

I still got this warning while I built the binary. I dont know if this highlights an issue but the binary ran successfully.
```
Warning: The specified HIP target: gfx902 is unknown. Correct compilation is not guaranteed.
```

mattsinc commented:
> I made the new image locally and built the square binary. I was able to successfully run the binary with apu_se.py with the new image.
> 
> I still got this warning while I built the binary. I dont know if this highlights an issue but the binary ran successfully.
> 
> ```
> Warning: The specified HIP target: gfx902 is unknown. Correct compilation is not guaranteed.
> ```

You can ignore that warning.  Essentially it's ROCms way of saying the code is not well optimized for APUs and thus best performance on an APU (like gfx902) is not guaranteed.

Harshil2107 commented:
I am trying to build the pannotia resource files but I can not build it.

I use the following command
```
docker run --rm -v ${PWD}:${PWD} -w ${PWD} -u $UID:$GID ghcr.io/gem5/gcn-gpu:v24-0 make gem5-fusion
```
I have also updated the line 11 GEM5_PATH to point to gem5 on my system. 

I get the following error
```
make -f Makefile.gem5-fusion
make[1]: *** No rule to make target '../graph_parser/util.cpp', needed by 'bin/bc.gem5'.  Stop.
make[1]: Entering directory '/home/harshilp/work/gem5-resources/src/gpu/pannotia/bc'
make[1]: Leaving directory '/home/harshilp/work/gem5-resources/src/gpu/pannotia/bc'
make: *** [Makefile:8: gem5-fusion] Error 2
```

abmerop commented:
This was never possible.  Docker refuses to pull from anything outside of the --volume so it cannot reference the m5ops.h file it wants.

Still a TODO is to have a discussion on how to work around this.  The primary issue being because gem5 and gem5-resources are different repos I do not think we can make any assumptions about the locations of both repos as would be required for docker's pathing.


Edit: Sorry, got ahead of myself. For the graph parse simply changing `-v ${PWD}:${PWD}` to `-v ${PWD}/..:${PWD}/..` will work, but then you will run into the m5ops issue above.

mattsinc commented:
> This was never possible. Docker refuses to pull from anything outside of the --volume so it cannot reference the m5ops.h file it wants.
> 
> Still a TODO is to have a discussion on how to work around this. The primary issue being because gem5 and gem5-resources are different repos I do not think we can make any assumptions about the locations of both repos as would be required for docker's pathing.

@abmerop is right that there are issues with m5ops here, but I don't think that is the issue @Harshil2107 is facing (the error here is with util.cpp ... I believe m5ops will be the next error after this one is fixed).  The error appears to be what directory Harshil is in and what volume he is using.  @abmerop is right we need to update the directions for this.

@Harshil2107 I am not at a terminal at the moment so I can't send a specific command, but I recommend you set your volume (-v) to be a shared directory that contains both gem5 and gem5-resources.  Then either a) set your working directory (-w) to be the BC folder or b) use a cd command at the end of the line to cd to the BC directory followed by a semicolon followed by make gem5-fusion.

I also see @abmerop has updated his comment while I was typing this saying something similar.  My suggestion here should resolve both m5ops and util.cpp

Harshil2107 commented:
Thank you. I was able to build the binary. Should the documentation be updated to make it more clear on how to build this binary?


 I will build the binaries, upload them to cloud and create JSONs to add to mongoDB resources database.

Harshil2107 commented:
I have built the binaries they are located at:
https://storage.googleapis.com/dist.gem5.org/dist/v24-0/test-progs/halo-finder/ForceTreeTest
https://storage.googleapis.com/dist.gem5.org/dist/v24-0/test-progs/heterosync/allSyncPrims-1kernel
https://storage.googleapis.com/dist.gem5.org/dist/v24-0/test-progs/hip-samples/2dshfl
https://storage.googleapis.com/dist.gem5.org/dist/v24-0/test-progs/hip-samples/MatrixTranspose
https://storage.googleapis.com/dist.gem5.org/dist/v24-0/test-progs/hip-samples/dynamic_shared
https://storage.googleapis.com/dist.gem5.org/dist/v24-0/test-progs/hip-samples/inline_asm
https://storage.googleapis.com/dist.gem5.org/dist/v24-0/test-progs/hip-samples/sharedMemory
https://storage.googleapis.com/dist.gem5.org/dist/v24-0/test-progs/hip-samples/shfl
https://storage.googleapis.com/dist.gem5.org/dist/v24-0/test-progs/hip-samples/stream
https://storage.googleapis.com/dist.gem5.org/dist/v24-0/test-progs/hip-samples/unroll
https://storage.googleapis.com/dist.gem5.org/dist/v24-0/test-progs/lulesh/lulesh
https://storage.googleapis.com/dist.gem5.org/dist/v24-0/test-progs/pannotia/bc.gem5
https://storage.googleapis.com/dist.gem5.org/dist/v24-0/test-progs/pannotia/color_max.gem5
https://storage.googleapis.com/dist.gem5.org/dist/v24-0/test-progs/pannotia/fw_hip.gem5
https://storage.googleapis.com/dist.gem5.org/dist/v24-0/test-progs/pannotia/mis_hip.gem5
https://storage.googleapis.com/dist.gem5.org/dist/v24-0/test-progs/pannotia/pagerank.gem5
https://storage.googleapis.com/dist.gem5.org/dist/v24-0/test-progs/pannotia/sssp.gem5

https://storage.googleapis.com/dist.gem5.org/dist/v24-0/test-progs/pennant/pennant [ Note: I have tarred all the contents of the build directory, do correct me if I am wrong]

https://storage.googleapis.com/dist.gem5.org/dist/v24-0/test-progs/square/square

Edit: I also think we should use these full links instead of the http one as these links would not throw the ssl certificate error.

mattsinc commented:
> Thank you. I was able to build the binary. Should the documentation be updated to make it more clear on how to build this binary?
> 
> I will build the binaries, upload them to cloud and create JSONs to add to mongoDB resources database.

I can't find the chain at the moment, but this has been a persistent issue @abmerop , @BobbyRBruce , and I have been discussing about how best to update the documentation.  Especially for Pannotia as all of them are similar in concept, but have small details that differ.  The simplest way, but maybe not the best way, to update things currently would be to update each Pannotia (and similarly for other workloads) documentation page to use something similar to what I sent you above.  I know @BobbyRBruce would prefer a single webpage though consolidating all Pannotia apps into a single page instead -- if we go that route, then we need to figure out how to express the differences for each app (e.g., how to run the different versions of Color/SSSP) and how to structure the files so that folks understand that while each Pannotia app has its own folder in gem5-resources, there is actually a single README that "rules them all" sitting above it.  Maybe a symlink would be the right solution?

I can get someone here to update these, or if you feel like you have a handle on it go ahead @Harshil2107 and I can review it.

Harshil2107 commented:
I will push a commit  with the updated links after I make the resources JSON and test them tommorrow.

abmerop commented:
Some updates...:
- Replaced pre-built paths with the https / googleapi URls
- Updated all gcn-gpu dockers to v24-0 from v23-1
- Added build instructions for pannotia apps (could use feedback on this)


For the pre-builts, I noticed the apps with different variants do not have anything built yet (color_maxmin, pagerank_spmv, and sssp_ell), so I left the line in the readme saying those would be added soon.

mattsinc commented:
> Some updates...:
> 
>     * Replaced pre-built paths with the https / googleapi URls
> 
>     * Updated all gcn-gpu dockers to v24-0 from v23-1
> 
>     * Added build instructions for pannotia apps (could use feedback on this)
> 
> 
> For the pre-builts, I noticed the apps with different variants do not have anything built yet (color_maxmin, pagerank_spmv, and sssp_ell), so I left the line in the readme saying those would be added soon.

@abmerop is this read for review, or are you waiting on anything?

abmerop commented:
This would be more complete if all the pre-built paths were there.  I don't think we can upload stuff to the website directly so I guess @Harshil2107  would need to do this?

Otherwise it is ready for review.  I originally wanted to add GPUFS targets / Makefiles, however ROCm 4.0 is too old to recognize the MI200 target and errors out (not a warning like gfx902). Therefore, I need to think of another plan for GPUFS Makefiles and will need additional README.md updates.  In the meantime though, the instructions for GPUSE are all included in this PR.

abmerop commented on src/gpu/pannotia/color/README.md:
````diff
@@ -6,63 +6,68 @@ tags:
 ```
-cd src/gpu/pannotia/clr
-docker run --rm -v ${PWD}:${PWD} -w ${PWD} -u $UID:$GID ghcr.io/gem5/gcn-gpu bash -c "export VARIANT=MAXMIN ; make gem5-fusion"
+docker run --rm -v ${PWD}:${PWD} -w ${PWD} -u $UID:$GID ghcr.io/gem5/gcn-gpu:v24-0 bash -c 'export VARIANT=MAXMIN ; cd gem5-resources/src/gpu/pannotia/color; make gem5-fusion'
````
Missing pre-built for this

Harshil2107 replied:
https://storage.googleapis.com/dist.gem5.org/dist/v24-0/test-progs/pannotia/color_maxmin.gem5

abmerop commented on src/gpu/pannotia/pagerank/README.md:
````diff
@@ -6,61 +6,68 @@ tags:
+To compile the "maxmin" variant from the gem5 directory:
+
+```
+docker run --rm -v ${PWD}:${PWD} -w ${PWD} -u $UID:$GID ghcr.io/gem5/gcn-gpu:v24-0 bash -c 'export VARIANT=SPMV ; cd gem5-resources/src/gpu/pannotia/pagerank; make gem5-fusion'
````
Missing pre-built for this

Harshil2107 replied:
https://storage.googleapis.com/dist.gem5.org/dist/v24-0/test-progs/pannotia/pagerank_spmv.gem5

abmerop commented on src/gpu/pannotia/sssp/README.md:
````diff
@@ -6,61 +6,68 @@ tags:
+To compile the "maxmin" variant from the gem5 directory:
+
+```
+docker run --rm -v ${PWD}:${PWD} -w ${PWD} -u $UID:$GID ghcr.io/gem5/gcn-gpu:v24-0 bash -c 'export VARIANT=ELL ; cd gem5-resources/src/gpu/pannotia/sssp; make gem5-fusion'
````
Missing pre-built for this

Harshil2107 replied:
https://storage.googleapis.com/dist.gem5.org/dist/v24-0/test-progs/pannotia/sssp_ell.gem5

abmerop reviewed (commented):
To be more specific, these are the pre-builts currently missing:

abmerop commented:
Thanks @Harshil2107 .  @mattsinc This should be completely ready now.

mattsinc commented on README.md:
````diff
@@ -335,52 +335,26 @@ To compile:
 
 ```
-git clone https://gem5.googlesource.com/public/gem5-resources
+git clone https://github.com/gem5/gem5
````
I think this should be https://github.com/gem5/gem5-resources instead of https://github.com/gem5/gem5?

abmerop replied:
Fixed

mattsinc commented on README.md:
```diff
@@ -335,52 +335,26 @@ To compile:
-Custom commands can be added to the command processor in gem5 to control
-the GPU in novel ways.
-
-## Compilation
```
I don't think the Compilation header should have been removed?  Or was it never in square?

abmerop replied:
This is for the hsa-agent-pkt benchmark that I removed

mattsinc commented on README.md:
```diff
@@ -397,26 +371,26 @@ docker run --rm -v ${PWD}:${PWD} -w ${PWD} -u $UID:$GID ghcr.io/gem5/gcn-gpu mak
 Individual programs can be made by specifying the name of the program
 
-By default, this code builds for gfx801, a GCN3-based APU. This can be
+By default, this code builds for gfx902, a VEGA-based APU. This can be
```
I did not see where this flag was used -- is HCC_AMDGPU_TARGET still correct?  HCC is gone now, right?

abmerop replied:
This in the docker image now.  Previously was in the HACC docker which essentially just defined some environment variables.  That was combined with the gcn-gpu docker image

mattsinc commented on src/gpu/pannotia/bc/Makefile.gem5-fusion:
```diff
@@ -8,7 +8,7 @@ HIP_PATH ?= /opt/rocm/hip
 
 # these are needed for m5ops
-GEM5_PATH ?= /path/to/gem5
+GEM5_PATH ?= ../../../../../
```
I won't fight this hard, but I do think the prior default at least gave users something clear about what needed to be changed, whereas this seems more like a "magic" path

abmerop replied:
I added a comment explaining what this is as a compromise.  All of the README.md's right now mention that the commands assume gem5-resources is checked out in the gem5 directory

mattsinc commented on src/gpu/pannotia/bc/README.md:
````diff
@@ -6,44 +6,50 @@ tags:
 ```
 
-If you use the Makefile.default file instead, the Makefile will generate code designed to run on the real GPU instead.  Moreover, note that Makefile.gem5-fusion requires you to set the GEM5_ROOT variable (either on the command line or by modifying the Makefile), because the Pannotia applications have been updated to use [m5ops](https://www.gem5.org/documentation/general_docs/m5ops/).  By default, ...
+Alternately from the gem5 directory, still assuming gem5-resources is checked out in the gem5 directory:
````
Alternatively?

abmerop replied:
Fixed in all places

mattsinc commented on src/gpu/pannotia/bc/README.md:
```diff
@@ -6,44 +6,50 @@ tags:
 ## Compilation and Running
 
-To compile BC:
+BC requires m5ops and common graph parsing libraries located in the parent directory. Docker requires that the paths to both are located within the --volume (-v) parameter and docker will not follow symlinks. The below instructions assume that gem5-resources is checked out in the gem5 directory. If that is not the case, please adapt your docker command with the correct paths. To compile BC:
```
Do we need a command telling them how to compile the m5ops?

abmerop replied:
I added another section in each README.md stating that m5ops needs to be built first and added a link to the website explaining how to do that

mattsinc commented on src/gpu/pannotia/fw/README.md:
````diff
@@ -6,43 +6,50 @@ tags:
 ```
 
-If you use the Makefile.default file instead, the Makefile will generate code designed to run on the real GPU instead.  Moreover, note that Makefile.gem5-fusion requires you to set the GEM5_ROOT variable (either on the command line or by modifying the Makefile), because the Pannotia applications have been updated to use [m5ops](https://www.gem5.org/documentation/general_docs/m5ops/).  By default, ...
+Alternately from the gem5 directory, still assuming gem5-resources is checked out in the gem5 directory:
````
I'm not going to comment on all, but same minor wording thing here

abmerop replied:
Fixed in all places

mattsinc reviewed (changes requested):
A few minor things, thanks for doing this!

mattsinc reviewed (approved):
Thanks @abmerop !
